components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx

```diff
@@ -88,6 +88,46 @@ export class Panel extends React.Component<Props, State> {
     console.log('doNothing click')
   }
 
+  onContributionAmountChange = (publisherKey: string, newAmount: string) => {
+    const newValue = parseInt(newAmount, 10)
+
+    if (newValue === 0) {
+      this.actions.removeRecurringContribution(publisherKey)
+    } else {
+      this.actions.saveRecurringDonation(publisherKey, newValue)
+    }
+  }
+
+  generateAmounts = () => {
+    const { rates } = this.props.rewardsPanelData.walletProperties
+
+    return [0, 1, 5, 10].map((value: number) => {
```

Comment:
this values needs to come from publisher list and default back to this if publisher didn't set it